util/resolver/resolver.go

```diff
@@ -149,16 +149,66 @@ func NewRegistryConfig(m map[string]config.RegistryConfig) docker.RegistryHosts
 	)
 }
 
-func New(ctx context.Context, hosts docker.RegistryHosts, sm *session.Manager) remotes.Resolver {
+type SessionAuthenticator struct {
+	sm      *session.Manager
+	groups  []session.Group
+	mu      sync.RWMutex
+	cache   map[string]credentials
+	cacheMu sync.RWMutex
+}
+
+type credentials struct {
+	user    string
+	secret  string
+	created time.Time
+}
+
+func NewSessionAuthenticator(sm *session.Manager, g session.Group) *SessionAuthenticator {
+	return &SessionAuthenticator{sm: sm, groups: []session.Group{g}, cache: map[string]credentials{}}
+}
+
+func (a *SessionAuthenticator) credentials(h string) (string, string, error) {
+	a.cacheMu.RLock()
+	c, ok := a.cache[h]
+	if ok && time.Since(c.created) < time.Minute {
```

Comment:
Where does `time.Minute` come from? Is that the expiration time of creds?

Reply:
Yes. I guess I could add a named constant for it so it is clearer.